runtime/pallene_core.h

```diff
@@ -57,20 +57,20 @@ int pallene_runtime_record_index_error(
     PALLENE_NORETURN;
 
 void pallene_runtime_array_metatable_error(
-    lua_State *L, int line)
+    lua_State *L, const char* file, int line)
     PALLENE_NORETURN;
 
 TString *pallene_string_concatN(
     lua_State *L, size_t n, TString **ss);
 
 void pallene_grow_array(
-    lua_State *L, Table *arr, unsigned int ui, int line);
+    lua_State *L, Table *arr, unsigned int ui, const char* file, int line);
```

Comment:
In some of the functions the file+line are the first arguments while in others they are the last. I think this might be because in the past the argument order reflected the order that these things appeared in the error message.

We could take this opportunity to make things more consistent by always putting the file+line at the start. (Since the pallene_runtime_tag_check_error is a vararg function, putting it at the end would not be an option in that case.)